### PR TITLE
IFTAQCAA-12 - test script to cover recently viewed products

### PR DESCRIPTION
**Pull request by dkrutenko**
Check if products are showing on the home page in recently viewed products

### CODE REVIEW

File: src/test/java/com/softserveinc/ita/SearchTest.java [hunk 1]
```diff
@@ -34,4 +34,33 @@ public void verifySearchResultsTest() {
...
+                .selectRandomSubCategory()
+                .selectFirstItemFromProductPage()
+                .getProductTitle();
+
+        var lastViewedItemFromMainPage = homePage
+                .clickOnMainPageLogo()
+                .homePageLastViewedProductTitle(expectedItem);
+
+        assertThat(lastViewedItemFromMainPage)
+                .as("Test failed: last viewed item must be: " + expectedItem)
```
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/SearchTest.java:55]: 'should be' instead of 'must be'

'should be' sounds more polite

**Review by Nikodych — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/pageobjects/HomePage.java [hunk 2]
```diff
@@ -40,22 +42,20 @@ public ProductPage openLastProductOfSection(String sectionName) {
         return new ProductPage();
     }
 
-    public String homePageLastViewedProductTitle(String expectedItem) {
+    public List<String> homePageLastViewedProductTitle(String expectedItem) {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/HomePage.java:45]: I mean something else, rename and do not pass anything, you just need from these screen return list of items titles

File: src/main/java/com/softserveinc/ita/pageobjects/HomePage.java [hunk 3]
```diff
@@ -40,22 +42,20 @@ public ProductPage openLastProductOfSection(String sectionName) {
...
                 .filter(text -> text.contains(expectedItem))
-                .collect(Collectors.toList())
-                .stream()
-                .findFirst()
-                .toString();
+                .collect(Collectors.toList());
     }
 
-    public ProductPage clickOnLastViewedItem() {
+    public ProductPage clickOnLastViewedItem(String expectedItem) {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/HomePage.java:54]: openLastViewedItemByTitle

File: src/main/java/com/softserveinc/ita/pageobjects/SearchResultPage.java [hunk 4]
```diff
@@ -19,4 +19,15 @@
                 .filter(text -> text.contains(productName))
                 .collect(toList());
     }
+
+    public ProductPage selectFirstItemFromProductPage() {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/SearchResultPage.java:23]: selectFirstItemFromProductList

File: src/main/java/com/softserveinc/ita/pageobjects/HomePage.java (hunk 2 above)
Comment (Nikodych) [reply src/main/java/com/softserveinc/ita/pageobjects/HomePage.java:45]: and do some magic in test

Comment (dkrutenko) [reply src/main/java/com/softserveinc/ita/pageobjects/HomePage.java:45]: you mean:
   public ElementsCollection selectFirstItemFromLastProduct() {
        return listOfLastViewedItems
                .shouldBe(sizeNotEqual(0), ofSeconds(8)); ???

and in test i have to stream and filter by name, right?

File: src/main/java/com/softserveinc/ita/pageobjects/HomePage.java [hunk 5]
```diff
@@ -42,16 +42,12 @@ public ProductPage openLastProductOfSection(String sectionName) {
         return new ProductPage();
     }
 
-    public List<String> homePageLastViewedProductTitle(String expectedItem) {
+    public ElementsCollection selectFirstItemFromLastProduct() {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/HomePage.java:45]:
```
List<String >getTitlesFromListOfLastViewedProducts

listOfLastViewedItems
                .shouldBe(sizeNotEqual(0), ofSeconds(8))
                .stream()
                .map(SelenideElement::text)
                 .collect(toList());
```

File: src/test/java/com/softserveinc/ita/SearchTest.java [hunk 6]
```diff
@@ -50,14 +51,17 @@ public void verifyRecentlyViewedProducts() {
 
         var lastViewedItemFromMainPage = homePage
```
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/SearchTest.java:52]: rename variable

File: src/test/java/com/softserveinc/ita/SearchTest.java [hunk 7]
```diff
@@ -50,14 +51,17 @@ public void verifyRecentlyViewedProducts() {
...
                 .clickOnMainPageLogo()
-                .homePageLastViewedProductTitle(expectedItem);
+                .selectFirstItemFromLastProduct();
 
-        assertThat(lastViewedItemFromMainPage)
+        assertThat(lastViewedItemFromMainPage
+                .stream()
+                .map(SelenideElement::text)
+                .filter(text -> text.contains(expectedItem)))
```
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/SearchTest.java:59]: rather you will not need filtering here, just insert into assert the list and contains should find

**Review by Nikodych — CHANGES_REQUESTED** [review]

**Review by Nikodych — APPROVED** [review]

**Comment by Nikodych** [comment]
need to sync with master

**Comment by dkrutenko** [comment]
merged

**Pull request merged**